src/imgui/queryBuilder.hpp

```diff
@@ -0,0 +1,152 @@
+#ifndef CASH_OVERFLOW_QUERYBUILDER_HPP
+#define CASH_OVERFLOW_QUERYBUILDER_HPP
+
+#include <algorithm>
+#include <optional>
+#include <string>
+#include <memory>
+#include <vector>
+
+#include "util.hpp"
+
+
+namespace cash_overflow::db {
+// Query::createQueryOfType("insert").into("user").with_value("jorge").on_column("username")
+//                                                .with_value("jndaiskjndad").on_column("password").go();
+// TODO: Return expected
+struct QueryBuilder
+{
+  virtual ~QueryBuilder(){};
+  QueryBuilder() = default;
+  virtual std::string go() = 0;
+};
+
+class SelectQueryBuilder;
+class InsertQueryBuilder;
+
+class Query
+{
+public:
+  friend class InsertQueryBuilder;
+  friend class SelectQueryBuilder;
+  static InsertQueryBuilder insert();
+  static SelectQueryBuilder select();
+
+private:
+  Query() = default;
+};
+
+
+class InsertQueryBuilder : public QueryBuilder
+{
+public:
+  InsertQueryBuilder &into(std::string name)
+  {
+    table = std::move(name);
+    return *this;
+  }
+  InsertQueryBuilder &on_column(std::string const &name)
+  {
+    columns.push_back(name);
+    return *this;
+  }
+  InsertQueryBuilder &with_value(std::string const &name)
+  {
+    values.push_back(name);
+    return *this;
+  }
+
+  std::string go() override
+  {
+    std::string v;
+    preparedQuery.append("insert into ");
```

Comment:
I don't know a lot about strings, but maybe consider std::ostringstream and use operator<< instead of append